skops/io/_numpy.py

```diff
@@ -15,22 +16,29 @@ def ndarray_get_state(obj, dst):
         "__module__": get_module(type(obj)),
     }
 
-    try:
-        f_name = f"{uuid4()}.npy"
-        with open(Path(dst) / f_name, "wb") as f:
-            np.save(f, obj, allow_pickle=False)
+    if obj.dtype != object:
+        try:
+            digest = joblib.hash(obj)
```

Comment:
I think it'd be nicer to do `id(obj)` which checks memory address. It's clearly much faster, and in our usecase it does prevent persisting identical arrays. pickle does the same.

Reply:
The problem with `id(obj)` is that it wouldn't catch exact copies, would it? Is speed even a concern here?

Reply:
yes, it doesn't catch exact copies, but I think we're more worried about storing the same object several times rather than storing copies. I'd say creating a hash of a large matrix could be a bottleneck, and I think sticking with pickle convention here would be nicer.

Reply:
I did the change to `id(obj)` but then the tests failed. The reason is that lines like those:

```python
        self.vector = X[0]
        self.vector_2 = X[0]
```

produce different ids. Is this the behavior we want?

Also, I checked the timing for hashing:

```python
>>> x = np.random.random((10000, 1000))
>>> %timeit joblib.hash(x)
112 ms ± 2.55 ms per loop (mean ± std. dev. of 7 runs, 10 loops each)
>>> y = np.random.random((10, 10))
53.5 µs ± 89.1 ns per loop (mean ± std. dev. of 7 runs, 10,000 loops each)
```

So both a few big arrays and many small arrays shouldn't be a big problem.

Reply:
Yes, I'm aware, the question is if we want to store two different copies of the array when it was indexed, even though they're identical.

Reply:
Yeah I don't think that's something we really have to optimize for here.